sebs/fission/fission.py

```diff
@@ -0,0 +1,373 @@
+import logging
+import os
+import shutil
+import subprocess
+import docker
+import json
+from time import sleep
+from typing import Dict, Tuple, List
+from sebs.faas.storage import PersistentStorage
+from sebs.cache import Cache
+from sebs.config import SeBSConfig
+from sebs.faas.function import Function
+from sebs.faas.system import System
+from sebs.fission.fissionFunction import FissionFunction
+from sebs.benchmark import Benchmark
+from sebs.fission.config import FissionConfig
+from sebs.fission.minio import Minio
+from tools.fission_preparation import (
+    check_if_minikube_installed,
+    run_minikube,
+    check_if_k8s_installed,
+    check_if_helm_installed,
+    stop_minikube,
+)
+
+
+class Fission(System):
+    language_image: str
+    language_builder: str
+    storage: Minio
+    httpTriggerName: str
+    functionName: str
+    packageName: str
+    envName: str
+    shouldCallBuilder: bool = False
+    _config: FissionConfig
+
+    def __init__(
+        self,
+        sebs_config: SeBSConfig,
+        config: FissionConfig,
+        cache_client: Cache,
+        docker_client: docker.client,
+    ):
+        super().__init__(sebs_config, cache_client, docker_client)
+        self._added_functions: List[str] = []
+        self._config = config
+
+    @staticmethod
+    def name():
+        return "fission"
+
+    @property
+    def config(self) -> FissionConfig:
+        return self._config
+
+    @staticmethod
+    def add_port_forwarding(port=5051):
+        podName = (
+            subprocess.run(
+                f"kubectl --namespace fission get pod -l svc=router -o name".split(),
+                stdout=subprocess.PIPE,
+            )
+            .stdout.decode("utf-8")
+            .rstrip()
+        )
+        subprocess.Popen(
```

Comment:
I think we can run `subprocess.run`, it's easier to use. Also, we defined somewhere a wrapper for `subprocess.run` that automatically checks for errors, throws exceptions, and decodes output.